### PR TITLE
added definiton of chunk and referenced it in the docs

### PR DESCRIPTION
**Pull request by Biswajitghosh98**
Fixes https://github.com/thanos-io/thanos/issues/3627

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

- Update store.md
- Update design.md
- Update storage.md
- Update troubleshooting.md

### CODE REVIEW

**Comment by Biswajitghosh98** [comment]
cc @GiedriusS 
Please review the changes 😃

File: docs/design.md [hunk 1]
```diff
@@ -63,24 +63,26 @@ The meta.json is updated during upload time on sidecars.
               │                   Object Storage                 │
               └──────────────────────────────────────────────────┘
 ```
+#### Note
```
Comment (kakkoyun) [inline docs/design.md:66]: So add definitions for chunk and chunk file and please link them accordingly.

**Review by kakkoyun — CHANGES_REQUESTED** [review]
Looking good. Let's fix it as we discussed in the meeting.

File: docs/design.md (hunk 1 above)
Comment (Biswajitghosh98) [reply docs/design.md]: @kakkoyun Done !

**Review by bwplotka — APPROVED** [review]
Thanks 🤗

**Pull request merged**